modules/thermo-cycler/thermo-cycler-arduino/gcode.cpp

```diff
@@ -0,0 +1,202 @@
+#include "gcode.h"
+
+#undef GCODE_DEF
+#define GCODE_DEF(_, str) #str
+
+const String COMMAND_CODES[] = { GCODES_TABLE };
+const char _CHARACTERS_TO_STRIP[] = {' ', '\r', '\n'};
+
+GcodeHandler::GcodeHandler() {}
+
+void GcodeHandler::_strip_serial_buffer()
+{
+  char temp[_serial_buffer_string.length()];
+  char * p;
+  p = temp;
+  bool skip = false;
+  for (const auto& sb_char : _serial_buffer_string)
+  {
+    for (const auto& c: _CHARACTERS_TO_STRIP)
+    {
+      if (sb_char == c)
+      {
+        skip = true;
+      }
+    }
+    if (!skip)
+    {
+      *p = sb_char;
+      p++;
+    }
+    skip = false;
+  }
+  *p = '\0';
+  _serial_buffer_string = temp;
+}
+
+/* Searches for gcode in the given string and updates the index reference with
+ * index of gcode found */
+bool GcodeHandler::_find_command(String _string, uint8_t * code_int, uint8_t * index)
```

Comment:
```suggestion
bool GcodeHandler::_find_command(const String& _string, uint8_t * code_int, uint8_t * index)
```

This way you don't copy the string in every time